src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -1017,10 +1033,14 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger? errorLogger, Cancella
                 {
                     // At this point we have a compilation with nothing yet computed. 
                     // We pass it to the generators, which will realize any symbols they require. 
-                    compilation = RunGenerators(compilation, Arguments.ParseOptions, generators, analyzerConfigProvider, additionalTextFiles, diagnostics);
+                    (compilation, var timingInfo) = RunGenerators(compilation, Arguments.ParseOptions, generators, analyzerConfigProvider, additionalTextFiles, Arguments.ReportAnalyzer, diagnostics);
 
                     bool hasAnalyzerConfigs = !Arguments.AnalyzerConfigPaths.IsEmpty;
                     bool hasGeneratedOutputPath = !string.IsNullOrWhiteSpace(Arguments.GeneratedFilesOutputDirectory);
+                    if (Arguments.ReportAnalyzer)
```

Comment:
nit: do we need this condition? We could do `(compilation, generatorTimingInfo) = RunGenerators(...);` instead above (line 1036).

Reply:
That doesn't quite work because `RunGenerators` returns a plain old `struct`, not a nullable one. Let me see if I can change that though.

Reply:
That wasn't too hard to change.